### DISCUSSION TITLE
Transfer of a file content to the render function

### DISCUSSION THREAD

**Discussion by Linuxmaker** [chillerlan/php-qrcode#154] [General]
Hello,
Your tool is fantastic. However, I am looking for a way to transfer a file to the QRCode object instead of a string. The background is the bank QR code with 32 lines, the information for which is in the variable $data.
`$qrcode = (new QRCode($options))→render('https://www.google.com');`

In the class definition QRCode.php it says from line 179:
```
public function render(string $data, string $file = null){
return $this->initOutputInterface($data)->dump($file);
}
```

How could this be changed so that a file with 32 lines of text can be passed instead?

Many greetings
Linuxmaker

**Comment by codemasher** [comment]
Hey, what do you mean by "transfer a file to the QRCode"? Do you want the content of that file in the QRCode? In that case you would just fetch the content of that file via `file_get_contents()`, like so:
```php
$qrcode = (new QRCode($options))->render(file_get_contents('/path/to/your/file.ext'));
```
The code snippet you posted manages the *output* of the finalized QRCode into a file.

**Reply by Linuxmaker** [reply]
Thanks for your response. Unfortunately, I get an exception regarding the allowed length:


> Fatal error: Uncaught chillerlan\QRCode\Data\QRCodeDataException: code length overflow. (1460 > 864 bit) in /var/www/html/php-qr-code-generator/vendor/chillerlan/php-qrcode/src/Data/QRDataAbstract.php:183 Stack trace: #0 /var/www/html/php-qr-code-generator/vendor/chillerlan/php-qrcode/src/Data/QRDataAbstract.php(95): chillerlan\QRCode\Data\QRDataAbstract->writeBitBuffer() #1 /var/www/html/php-qr-code-generator/vendor/chillerlan/php-qrcode/src/Data/QRDataAbstract.php(77): chillerlan\QRCode\Data\QRDataAbstract->setData() #2 /var/www/html/php-qr-code-generator/vendor/chillerlan/php-qrcode/src/QRCode.php(227): chillerlan\QRCode\Data\QRDataAbstract->__construct() #3 /var/www/html/php-qr-code-generator/vendor/chillerlan/php-qrcode/src/QRCode.php(194): chillerlan\QRCode\QRCode->initDataInterface() #4 /var/www/html/php-qr-code-generator/vendor/chillerlan/php-qrcode/src/QRCode.php(250): chillerlan\QRCode\QRCode->getMatrix() #5 /var/www/html/php-qr-code-generator/vendor/chillerlan/php-qrcode/src/QRCode.php(180): chillerlan\QRCode\QRCode->initOutputInterface() #6 /var/www/html/php-qr-code-generator/public/index.php(21): chillerlan\QRCode\QRCode->render() #7 {main} thrown in /var/www/html/php-qr-code-generator/vendor/chillerlan/php-qrcode/src/Data/QRDataAbstract.php on line 183

In QRDataAbstract.phpact.php, row 183:

```
if($this->bitBuffer->getLength() > $MAX_BITS){ 
                        throw new QRCodeDataException(sprintf('code length overflow. (%d > %d bit)', $this->bitBuffer->getLength(), $MAX_BITS)); 
}
``` 
Where can I increase the allowed length of the string in your program?

**Reply by codemasher** [reply]
The allowed content length is dictated by the version number (1-40). You can try to increase the version number or set it to `Version::AUTO`.

**Reply by Linuxmaker** [reply]
Thanks, 
with this values it runs
```
$options = new QROptions(
  [
    'eccLevel' => QRCode::ECC_L,
    'outputType' => QRCode::OUTPUT_MARKUP_SVG,
    'version' => 8,
  ]
);
```
instead "5".
